Mention using PINGs to keep tunnels alive (#75)
tfpauly/quic-proxy: draft-pauly-masque-quic-proxy.md

or merely forwarded from the target. The client replaces the Virtual Client
Connection ID with the real Client Connection ID before processing the packet further.

## Connection Maintenance in Forwarded Mode

When a client and proxy are using forwarded mode, it is possible that there can be

> long periods of time in which no stream data or datagram frames is exchanged

Reviewer: ```suggestion
long periods of time in which no ack-eliciting packets (see {{Section 2 of !QUIC-RETRANSMISSION=RFC9002}}) are exchanged
```

Reviewer 2: I like this as well but was too lazy to offer it as a third suggestion

> long periods of time in which no stream data or datagram frames is exchanged

Reviewer 2: ```suggestion
long periods of time in which no stream data or datagrams are exchanged
```

alternatively maybe just STREAM or DATAGRAM frames

between the client and proxy. If these periods extend beyond the max_idle_timeout

> values for the QUIC connection to the proxy (see {{Section 10.1 of QUIC}}),

Reviewer: ```suggestion
values for the client-to-proxy QUIC connection (see {{Section 10.1 of QUIC}}),
```

> values for the QUIC connection to the proxy (see {{Section 10.1 of QUIC}}),

Reviewer 2: Rather than refer to the TP, probably better to mention the effective idle timeout for the connection

Author: Done!

the QUIC connection might be closed by the proxy if the proxy does not use

> forwarded packets as an explicit liveness signal. To avoid this, clients SHOULD

Reviewer: Why not in the other direction too?

Author: I guess the proxy _could_, but if it's not tracking activity, it isn't aware that the client actually isn't idle (and if it is tracking, it doesn't need the keep alive). The client is the only one who knows about all of the connections.

Reviewer: Can you elaborate? What's preventing the client from timing out?

Author: The client is sending forwarded packets and it knows it is sending them actively — if it times out, that is the client's own problem (and it can just send PINGs to keep stuff alive itself).

send keepalive packets to the proxy before the idle timeouts would be reached,

> which can be done using PING frames.

Reviewer 2: The intent is good, but maybe you just want to defer to QUIC's liveness testing text (https://datatracker.ietf.org/doc/html/rfc9000#section-10.1.1) as to how exactly to do it?

Author: ```suggestion
which can be done using a PING frame or another ack-eliciting frame as described
in {{Section 10.1.1 of QUIC}}.
```

or merely forwarded from the target. The client replaces the Virtual Client
Connection ID with the real Client Connection ID before processing the packet further.

## Connection Maintenance in Forwarded Mode

When a client and proxy are using forwarded mode, it is possible that there can be
long periods of time in which no ack-eliciting packets (see {{Section 2 of !QUIC-RETRANSMISSION=RFC9002}}) are exchanged

> between the client and proxy. If these periods extend beyond the max_idle_timeout
> values for the client-to-proxy QUIC connection (see {{Section 10.1 of QUIC}}),

Author: ```suggestion
between the client and proxy. If these periods extend beyond the effective idle
timeout for the client-to-proxy QUIC connection (see {{Section 10.1 of QUIC}}),
```
